incydr/cli/__init__.py

```diff
@@ -1,21 +1,71 @@
+import os
+
 import click
 from rich.console import Console
 
-from incydr import Client
+from incydr.cli.utils import get_user_project_path
+
+ERROR_LOG_FILE_NAME = "incydr_cli.log"
 
 console = Console()
 
+
+def log_level_callback(ctx, param, value):
+    """callback to update log_level environment variable"""
+    if not value:
+        return
+    os.environ.update({"INCYDR_LOG_LEVEL": value})
+    return value
+
+
+def log_file_callback(ctx, param, value):
+    """callback to update log_file and log_stderr environment variable"""
+    if not value:
+        return
+    os.environ.update({"INCYDR_LOG_FILE": value})
+    return value
+
+
+def log_stderr_callback(ctx, param, value):
+    """callback to update log_stderr environment variable"""
+    if not value:
+        os.environ.update({"INCYDR_LOG_STDERR": "FALSE"})
+        return
+    os.environ.update({"INCYDR_LOG_STDERR": "TRUE"})
+    return value
+
+
+# TODO: do we want these options to be hidden from --help (add hidden=True to each def)
+
 log_level_option = click.option(
-    "--log-level", help="Set level for Incydr client logging.", default="WARNING"
+    "--log-level",
+    help="Set level for Incydr client logging.",
+    default="WARNING",
+    callback=log_level_callback,
+    expose_value=False,
 )
 
 log_file_option = click.option(
-    "--log-file", help="Specify file path to write log output to.", default=None
+    "--log-file",
+    help="Specify file path to write log output to.",
+    default=str(os.path.join(get_user_project_path("log"), ERROR_LOG_FILE_NAME)),
+    callback=log_file_callback,
+    expose_value=False,
+)
+
+log_stderr_option = click.option(
```

Comment:
Adds a `--log-stderr` flag for each command.  Defaults to `False` so that the console will only print out the single line error message.  If set to `True` then the full error will be output along with a stacktrace if applicable.  

Definitely looking for feedback on that default behavior

Reply:
Yeah I think this as default works.